pkg/messaging/pubsub/client.go

```diff
@@ -4,40 +4,29 @@ import (
 	"context"
 	"log"
 
-	"go.nanomsg.org/mangos/v3"
 	"go.nanomsg.org/mangos/v3/protocol"
 	"go.nanomsg.org/mangos/v3/protocol/sub"
 	_ "go.nanomsg.org/mangos/v3/transport/all"
-	"nodemon/pkg/entities"
 	"nodemon/pkg/messaging"
 )
 
-func subscribeToAlerts(socket protocol.Socket) error {
-	for alertType := range entities.AlertTypes {
-		err := socket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-func StartMessagingClient(ctx context.Context, nanomsgURL string, bot messaging.Bot) error {
-	socket, err := sub.NewSocket()
+func StartPubSubMessagingClient(ctx context.Context, nanomsgURL string, bot messaging.Bot) error {
+	pubSubSocket, err := sub.NewSocket()
 	if err != nil {
 		log.Printf("failed to get new sub socket: %v", err)
 		return err
 	}
-	defer func(socketPair protocol.Socket) {
-		if err := socketPair.Close(); err != nil {
+	defer func(pubSubSocket protocol.Socket) {
+		if err := pubSubSocket.Close(); err != nil {
 			log.Printf("Failed to close pair socket: %v", err)
 		}
-	}(socket)
-	if err := socket.Dial(nanomsgURL); err != nil {
+	}(pubSubSocket)
+	bot.SetPubSubSocket(pubSubSocket)
```

Comment:
Does your bot handle situation when `pubSubSocket` is `nil`? Or bot internal socket field has been set to `nil`?

Reply:
According to the function sub.NewSocket(), the socket cannot be nil. There is even no point to check for an error, because this type of socket cannot return an error. 
![Screenshot from 2022-06-07 22-27-26](https://user-images.githubusercontent.com/41806871/172466094-68bee8c9-0c78-49e6-af8e-72160e97cf7a.png)